crates/rbuilder/src/integration/playground.rs

```diff
@@ -94,6 +96,26 @@ impl Playground {
             },
         }?;
 
+        let start = Instant::now();
+        loop {
+            if start.elapsed().as_secs() > 10 {
+                return Err(PlaygroundError::Timeout);
+            }
+
+            // from the log file, check if the server has started
+            // by checking if the string "RPC server job: started" is present
+            let mut file = File::open(&log_path).map_err(|_| PlaygroundError::SetupError)?;
+            let mut contents = String::new();
+            file.read_to_string(&mut contents)
+                .map_err(|_| PlaygroundError::SetupError)?;
+
+            if contents.contains("RPC server job: started") {
+                break;
+            }
+
+            thread::sleep(std::time::Duration::from_millis(100));
+        }
```

Comment:
Something like this would be more idiomatic and avoid spamming io ops https://github.com/jmagnuson/linemux. I wouldn't call it a "must have" though.

Reply:
Yeah, I was hoping to do an http request to check some health check endpoint. But could not make it work.